src/app/public/theming/theme-class.directive.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  Directive,
+  ElementRef,
+  Input,
+  OnDestroy,
+  Renderer2
+} from '@angular/core';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  SkyThemeSettings
+} from './theme-settings';
+
+import {
+  SkyThemeService
+} from './theme.service';
+
+type SkyThemeClassMap = {[key: string]: string};
+
+/**
+ * Component to add classes conditionally based on the current theme.
+ *
+ * If the directive is within a `skyTheme` directive, it uses settings from that directive.
+ */
+@Directive({
+  selector: '[skyThemeClass]'
+})
+export class SkyThemeClassDirective implements OnDestroy {
+  private currentTheme: SkyThemeSettings | undefined;
+  private ngUnsubscribe = new Subject();
+  private initialClasses: string[] = [];
+  private skyThemeClassMap: SkyThemeClassMap;
+
+  constructor(
+    private ngEl: ElementRef,
+    private renderer: Renderer2,
+    private themeSvc: SkyThemeService
+  ) {
+    this.themeSvc.settingsChange
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe(settingsChange => {
+        this.themeSettings = settingsChange.currentSettings;
+      });
+  }
+
+  public ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  /**
+   * The HTML class attribute.
+   *
+   * @param value
+   */
+  @Input('class')
+  set className(value: string) {
```

Comment:
Code style > properties should come before the constructor. Only your class methods should be below the constructor.
Also, typescript defaults properties to `public`, but we try to always define scope - would you mind adding `public` to this?